brouter-routing-app/src/main/java/btools/routingapp/BRouterActivity.java

```diff
@@ -730,4 +759,123 @@ public void onRequestPermissionsResult (int requestCode, String[] permissions, i
+        }
+      } finally {
+        if (cursor != null)
+          cursor.close();
+      }
+      System.out.println("Brouter filename=" + filename + "\n file size=" + filesize);
+      scanResults.setText(scanResults.getText() + "File name=" + filename + "\nFile size=" + filesize + " bytes\n");
+      //  is the file extention ".brf" in the file name
+      if (filename.indexOf(".brf") != -1 && (filename.indexOf(".brf") == (filename.length() - 4))) {
+        System.out.println("Brouter  OK, file extention is .brf!!!");
+      } else {
+        System.out.println("Brouter  ERROR, please provide a file with the valid extention \".brf\"...");
+        // report error in UI and stop
+        scanResults.setText(scanResults.getText() + "ERROR: File extention must be \".brf\" \n");
+        return;
+      }
+      // profile size is generally < 30 kb, so set max size to 100 kb
+      if (filesize > 100000) {
+        System.out.println("Brouter  file size too big!!!");
+        // report error in UI and stop
+        scanResults.setText(scanResults.getText() + "ERROR: File size too big !!! \n");
+        return;
+      }
+
+      ContentResolver cr = getContentResolver();
+      String Profile_code = "";
+      try {
+        //    try to read the file
+        InputStream input = cr.openInputStream(dataUri);
+
+        BufferedReader reader = new BufferedReader(
+          new InputStreamReader(input));
+        Profile_code = reader.lines().collect(Collectors.joining(
+          System.getProperty("line.separator"))).toString();
+        System.out.println("Brouter Profile_CODE=" + Profile_code);
+
+        // consistency check
+        if (Profile_code.indexOf("highway=") == -1 || (Profile_code.indexOf("costfactor") == -1) || (Profile_code.indexOf("---context:global") == -1)) {
+          System.out.println("Brouter  ERROR, file content is not a valid profile for Brouter!, please provide a valid profile ...");
+          // report error in UI and stop
+          scanResults.setText(scanResults.getText() + "ERROR: this file is not a valid brouter-profile!!!! \n");
+          return;
+        }
+      } catch (IOException e) {
+        System.out.println(e);
+        // report error in UI and stop
+        scanResults.setText(scanResults.getText() + "ERROR: "  + e + "/n");
+      }
+
+      String baseDir = null;
+      InputStream configInput = null;
+      try
+      {
+        configInput = openFileInput( "config15.dat" );
+        BufferedReader br = new BufferedReader( new InputStreamReader( configInput ) );
+        baseDir = br.readLine();
+        System.out.println("Brouter  baseDir=" + baseDir );
+      }
+      catch (Exception e)
+      {
+        System.out.println("Brouter  exception by read config15.dat  " + e );
+        scanResults.setText(scanResults.getText() + "ERROR: " + e + " /n");
+      }
+      finally
+      {
+        if ( configInput != null ) try { configInput.close(); } catch (Exception ee) {}
+      }
```

Comment:
Use `ConfigHelper.getBaseDir(this)`

Reply:
I am not really familiar with brouter...
Sorry!